### ISSUE TITLE
Payload Nullified when the request type is 'DELETE'

### ISSUE THREAD

**Issue by alfonzjanfrithz** [openshift/openshift-restclient-java#359]
I notice that when deleting OpenShift Jobs from OpenShift web console it called:
  
| Name  | Value |
| ------------- | ------------- |
| URL  | https://$ENDPOINT/apis/batch/v1/namespaces/$NAMESPACE/jobs  |
| RequestMethod  | DELETE |
| Payload| {"kind":"DeleteOptions","apiVersion":"v1","propagationPolicy":"Foreground"} |

There is an inconsistency in the restclient API where it nullify any request with DELETE RequestMethod type. I understand some HTTP standards encourage to drop the payload on GET/DELETE although it is not explicitly stated here: https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html. 

https://github.com/openshift/openshift-restclient-java/blob/fee040f2872a406504a058fe67ab13ad4e061314/src/main/java/com/openshift/internal/restclient/DefaultClient.java#L316-L326

As much as I want to follow the common convention, I would prefer the GET/DELETE request method in this restclient not to drop the payload and let openshift endpoint to decide whether it will read the payload/ignore it. Otherwise, from openshift API side, the API needs to also follow the conventions by dropping the payload and open the parameter.

**Comment by adietish** [comment]
@jcantrill dare to comment, please?

**Comment by jcantrill** [comment]
Looks like we simply need a PR to correct line 320 and tests to verify.  If  DELETE call to kubernetes with a payload is a valid user call then we should support it.

**Comment by adietish** [comment]
@alfonzjanfrithz works for me, I'll drop a PR. Sorry that it took quite a bit to address this.

**Comment by adietish** [comment]
@alfonzjanfrithz we can allow a payload for **DELETE** but not for **GET** nor **HEAD** given that the http client library that we use won't allow it. See here: https://github.com/square/okhttp/blob/master/okhttp/src/main/java/okhttp3/internal/http/HttpMethod.java#L36
```
  public static boolean permitsRequestBody(String method) {
    return !(method.equals("GET") || method.equals("HEAD"));
  }
```

**Comment by danielallen88** [comment]
@adietish Any idea when this fix will be release? Thanks.

**Comment by adietish** [comment]
@danielallen88 this was released in 7.0.1-SNAPSHOT, current stable 7.1.1.Final should contain it. Isn't it?
